src/Sylius/Component/User/Security/Provider/VerificationTokenProvider.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\User\Security\Provider;
+
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\Component\User\Security\Generator\GeneratorInterface;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+class VerificationTokenProvider extends AbstractUniqueTokenProvider
+{
+    /**
+     * @param RepositoryInterface $userRepository
+     * @param GeneratorInterface $generator
+     * @param int $tokenLength
+     */
+    public function __construct(RepositoryInterface $userRepository, GeneratorInterface $generator, $tokenLength)
+    {
+        $this->userRepository = $userRepository;
+        $this->generator = $generator;
+        $this->tokenLength = (int) $tokenLength;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function isTokenUsed($token)
```

Comment:
I've never thought I will say it, but it actually looks like an overengineering :)